### PR TITLE
[https://nvbugs/5853720] [Fix] use cutedsl argmax only for fp32 dtype input

### PR DESCRIPTION
**Pull request by ameynaik-hub**
enable cutedsl based argmax only for fp32 dtype.

## Summary by CodeRabbit

* **Refactor**
  * Optimized argmax operation performance for float32 logits in speculative sampling workflows.

### CODE REVIEW

File: tensorrt_llm/_torch/speculative/interface.py [hunk 1]
```diff
@@ -548,7 +549,10 @@ def _draft_sampler_greedy(self, logits: torch.Tensor, d2t=None):
         Returns:
             draft_tokens: [num_tokens] - Sampled draft token ids (int32)
         """
-        draft_tokens = torch.argmax(logits, dim=-1)
+        if logits.dtype == torch.float32:
+            draft_tokens = argmax.argmax(logits)[:, 1].long()
```
[the remark this answers is not in the document]
Comment (ameynaik-hub) [reply tensorrt_llm/_torch/speculative/interface.py]: sorry let me add some background. Yes, GPTOSS uses FP32 that’s the only precision this kernel can support. However, for DeepSeek, the MTP draft uses BF16, which was causing problems. That’s why I added this.

[the remark this answers is not in the document]
Comment (dongfengy) [reply tensorrt_llm/_torch/speculative/interface.py]: I see thanks! Approved.

**Review by dongfengy — APPROVED** [review]

**Review by yweng0828 — APPROVED** [review]
LGTM

**Review by yweng0828 — APPROVED** [review]
